File: src/main/resources/output/IFC4X3_csharp/Entities/IfcRoot/IfcObjectDefinition/IfcTypeObject/IfcTypeProduct/IfcElementType/IfcDistributionElementType/IfcDistributionFlowElementType/IfcFlowMovingDeviceType/IfcFanType/IfcFanType.cs
public class IfcFanType : IfcFlowMovingDeviceType
{
    public IfcFanTypeEnum PredefinedType { get; set; }

    // === WHERE CLAUSES ===
    // CorrectPredefinedType : (PredefinedType <> IfcFanTypeEnum.USERDEFINED) OR ((PredefinedType = IfcFanTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcElementType.ElementType))
}

File: src/main/resources/output/IFC4X3_vbnet/Entities/IfcRoot/IfcObjectDefinition/IfcTypeObject/IfcTypeProduct/IfcElementType/IfcDistributionElementType/IfcDistributionFlowElementType/IfcFlowMovingDeviceType/IfcFanType/IfcFanType.vb
Public Class IfcFanType Inherits IfcFlowMovingDeviceType
    Public PredefinedType As IfcFanTypeEnum

    ' === WHERE CLAUSES ===
    ' CorrectPredefinedType : (PredefinedType <> IfcFanTypeEnum.USERDEFINED) OR ((PredefinedType = IfcFanTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcElementType.ElementType))
End Class
